src/main/webapp/app/course/competencies/competency-accordion/competency-accordion.component.ts

```diff
@@ -0,0 +1,167 @@
+import { Router } from '@angular/router';
+import { ICompetencyAccordionToggleEvent } from 'app/shared/competency/interfaces/competency-accordion-toggle-event.interface';
+import { CompetencyInformation, LectureUnitInformation, StudentMetrics } from 'app/entities/student-metrics.model';
+import { round } from 'app/shared/util/utils';
+import { Exercise } from 'app/entities/exercise.model';
+import dayjs from 'dayjs/esm';
+import { lectureUnitIcons, lectureUnitTooltips } from 'app/entities/lecture-unit/lectureUnit.model';
+
+@Component({
+    selector: 'jhi-competency-accordion',
+    templateUrl: './competency-accordion.component.html',
+    styleUrl: './competency-accordion.component.scss',
+})
+export class CompetencyAccordionComponent implements OnChanges {
+    @Input() course?: Course;
+    @Input() competency: CompetencyInformation;
+    @Input() metrics: StudentMetrics;
+    @Input() index: number;
+    @Input() openedIndex?: number;
+
+    @Output() accordionToggle = new EventEmitter<ICompetencyAccordionToggleEvent>();
+
+    open = false;
+    nextExercises: Exercise[] = [];
+    nextLectureUnits: LectureUnitInformation[] = [];
+
+    protected readonly faList = faList;
+    protected readonly faPdf = faFilePdf;
+    protected readonly lectureUnitIcons = lectureUnitIcons;
+    protected readonly lectureUnitTooltips = lectureUnitTooltips;
+    protected readonly getIcon = getIcon;
+    protected readonly getProgress = getProgress;
+    protected readonly getConfidence = getConfidence;
+    protected readonly getMastery = getMastery;
+    protected readonly round = round;
+
+    constructor(private router: Router) {}
+
+    ngOnChanges(changes: SimpleChanges) {
+        if (changes.openedIndex && this.index !== this.openedIndex) {
+            this.open = false;
+        }
+        if (changes.metrics) {
+            this.setNextExercises();
+            this.setNextLessonUnits();
+        }
+    }
+
+    setNextExercises() {
+        if (!this.metrics) {
+            this.nextExercises = [];
+        }
+
+        const submittedExercises = Object.keys(this.metrics.exerciseMetrics?.latestSubmission ?? {}).map(Number);
+        const competencyExercises = this.metrics.competencyMetrics?.exercises?.[this.competency.id] ?? [];
+        const nextExerciseInformations = competencyExercises
+            .filter((exerciseId) => !submittedExercises.includes(exerciseId))
+            .flatMap((exerciseId) => this.metrics.exerciseMetrics?.exerciseInformation?.[exerciseId] ?? [])
+            .filter((exercise) => exercise.startDate.isBefore(dayjs()) && exercise.dueDate.isAfter(dayjs()))
+            .sort((a, b) => a.dueDate.diff(b.dueDate));
+
+        // Workaround to convert ExerciseInformation to Exercise
+        this.nextExercises = nextExerciseInformations.map(
+            (exercise) =>
+                ({
+                    ...exercise,
+                    studentAssignedTeamIdComputed: exercise.studentAssignedTeamId,
```

Comment:
Why do we assign a number (exerciseInformation.studentAssignedTeamId) to a boolean (exercise.studentAssignedTeamId)? We should at least comment why we do this and why it is ok.